modules/nixpkgs.nix

```diff
@@ -1,26 +1,32 @@
-#
 # Nixpkgs module. The only exception to the rule.
-#
 # Provides a `pkgs` argument in `perSystem`.
-#
-# Arguably, this shouldn't be in flake-parts, but in nixpkgs.
-# Nixpkgs could define its own module that does this, which would be
-# a more consistent UX, but for now this will do.
-#
-# The existence of this module does not mean that other flakes' logic
-# will be accepted into flake-parts, because it's against the
-# spirit of Flakes.
-#
+topLevel@{ config, options, inputs, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib) mkPerSystemOption;
+
+in
 {
-  config = {
-    perSystem = { inputs', lib, ... }: {
-      config = {
-        _module.args.pkgs = lib.mkOptionDefault (
-          builtins.seq
-            (inputs'.nixpkgs or (throw "flake-parts: The flake does not have a `nixpkgs` input. Please add it, or set `perSystem._module.args.pkgs` yourself."))
-            inputs'.nixpkgs.legacyPackages
-        );
-      };
-    };
+  imports = [ inputs.nixpkgs.flakeModules.default ];
```

Comment:
We shouldn't make assumptions about the `inputs`, especially not in the core of flake parts.